src/gui/SearchWidget.cpp

```diff
@@ -94,10 +95,20 @@ bool SearchWidget::eventFilter(QObject* obj, QEvent* event)
             emit escapePressed();
             return true;
         } else if (keyEvent->matches(QKeySequence::Copy)) {
-            // If Control+C is pressed in the search edit when no text
-            // is selected, copy the password of the current entry.
+            // If the system Copy shortcut (typically Ctrl+C or Cmd+C) is pressed
+            // in the search edit when no text is selected, route the event to the
+            // main window. With the default shorcut configuration, this will copy
+            // the password of the current entry to the clipboard.
             if (!m_ui->searchEdit->hasSelectedText()) {
-                emit copyPressed();
+                // Prevent infinite recursion, in case the main window ends up
+                // sending this event back to us. This hasn't actually been observed
+                // in practice and is just a precaution.
+                if (m_sendingCopyShortcutEvent) {
+                    return true;
+                }
```

Comment:
This could be made a static function variable (eg, `static bool sendingCopyShortcutEvent;`)

Reply:
Ok, I've done that.